kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+
+public class ProcurementCardCreateDocumentServiceImpl extends org.kuali.kfs.fp.batch.service.impl.ProcurementCardCreateDocumentServiceImpl {
+
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ProcurementCardCreateDocumentServiceImpl.class);
+    
+    private GroupService procurementCardGroupService;
+    
+    public GroupService getProcurementCardGroupService(){
+    	return procurementCardGroupService;
+    }
+    
+    public void setProcurementCardGroupService(GroupService procurementCardGroupService){
+    	this.procurementCardGroupService = procurementCardGroupService;
+    }
+
+    public static final String REROUTE_PCDO_DOCUMENTS_IND_PARM_NM = "REROUTE_PCDO_DOCUMENTS_IND";
+    public static final String RECONCILER_GROUPS_TO_REROUTE_PARM_NM = "RECONCILER_GROUPS_TO_REROUTE";
+    
+	private boolean hasReconciler(ProcurementCardDefault procurementCardDefault) {
+        boolean retCode = true;
+        if (ObjectUtils.isNull(procurementCardDefault) ||
+            ObjectUtils.isNull(procurementCardDefault.getReconcilerGroupId()) ||
+            ObjectUtils.isNull(procurementCardDefault.getCardHolderSystemId())) {
+            retCode = false;
+        }
+        else {
+            List<String> groupMembers = new ArrayList<String>();
+            String reconcilerGroupId = procurementCardDefault.getReconcilerGroupId();
+            groupMembers = SpringContext.getBean(GroupService.class).getMemberPrincipalIds(reconcilerGroupId);
+            if (groupMembers.isEmpty() ||
+                (groupMembers.size() == 1 &&
+                 groupMembers.get(0).equals(procurementCardDefault.getCardHolderSystemId()))) {
+                retCode = false;
+            }
+        }
+        return retCode;
+    }
+	
+    /**
+     * This method looks for ProcurementCardDocuments that are in route status at the "AccountFullEdit" route node and routed to the error account FO. 
+     * Then checks for a valid reconciler and if found, reroutes the document back to the Reconciler.
+     * 
+     * 
+     * @return True if the routing was performed successfully.  A runtime exception will be thrown if any errors occur while routing.
+     * 
+     * @see org.kuali.kfs.fp.batch.service.ProcurementCardCreateDocumentService#rerouteProcurementCardDocuments(java.util.List)
+     */
+	@Override
+    public boolean rerouteProcurementCardDocuments() {
+        
+        // check for reconciler group rerouting
+        boolean groupRerouting = parameterService.getParameterValueAsBoolean(ProcurementCardRerouteDocumentsStep.class, REROUTE_PCDO_DOCUMENTS_IND_PARM_NM);
+        
+        Collection<ProcurementCardDocument> documents = null;
+        documents = retrieveUAProcurementCardDocumentsToRoute(KewApiConstants.ROUTE_HEADER_ENROUTE_CD);
+        
+        if ( LOG.isInfoEnabled() ) {
+            LOG.info("PCards to Reroute: " + new Integer(documents.size()).toString());
```

Comment:
The Kuali convention is to default to INFO, using string concatenation and checking for levels 'debug' and below to avoid unnecessary string concatenation evaluation:
https://kualico.atlassian.net/wiki/display/FINDOC/Java+Coding+Standards#JavaCodingStandards-Logging

That said, checking for .isInfoEnabled is unnecessary, as well as the triple conversion to get the 'int' into the String. I'll take another look at the logging statements in this class with that lens.